components/resourceDetails/resourceSharing/agentAccessList/index.jsx

```diff
@@ -32,7 +34,7 @@ const useStyles = makeStyles((theme) => createStyles(styles(theme)));
 export const TESTCAFE_ID_AGENT_ACCESS_LIST_SHOW_ALL =
   "agent-access-list-show-all";
 
-function AgentAccessList({ onLoading }) {
+function AgentAccessList({ onLoading, ...buttonProps }) {
```

Comment:
It's a bit weird that the rest of the properties for `AgentAccessList` is called `buttonProps` - the name of the component indicates that it's a list, after all. Further, the buttonProps is only used for a specific part of the component, which might cause confusion if we need to extend other elements.

I suggest adding buttonProps as proper property, and rename it to be more specific to how it's applied, e.g. `({ accessButtonProps, onLoading })`.